js/apps/admin-ui/src/util.ts

```diff
@@ -171,3 +171,18 @@ export const addTrailingSlash = (url: string) =>
   url.endsWith("/") ? url : url + "/";
 
 export const generateId = () => Math.floor(Math.random() * 1000);
+
+type PathParam = { [key: string]: string | null };
+
+export function generateUnencodedPath<Path extends string>(
+  originalPath: Path,
+  params: PathParam,
+): string {
+  const encodedParams: PathParam = {};
```

Comment:
Nice, I like your approach of using an empty object better than the suggestion I had.